gst/nnstreamer/nnstreamer_watchdog.c

```diff
@@ -0,0 +1,108 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * NNStreamer watchdog
+ * Copyright (C) 2024 Gichan Jang <gichan2.jang@samsung.com>
+ */
+/**
+ * @file	nnstreamer_watchdog.c
+ * @date	30 Oct 2024
+ * @brief	NNStreamer watchdog to manage the schedule in the element.
+ * @see		https://github.com/nnstreamer/nnstreamer
+ * @author	Gichan Jang <gichan2.jang@samsung.com>
+ * @bug		No known bugs except for NYI items
+ */
+
+#include "nnstreamer_watchdog.h"
+
+/**
+ * @brief Structure for NNStreamer watchdog.
+ */
+typedef struct _NnstWatchdog
+{
+  GMainContext *main_context;
+  GMainLoop *main_loop;
+  GThread *thread;
+  GSource *source;
+} NnstWatchdog;
+
+/**
+ * @brief Watchdog thread.
+ */
+static gpointer
+_nnstreamer_watchdog_thread (gpointer ptr)
+{
+  NnstWatchdog *watchdog = (NnstWatchdog *) ptr;
+
+  g_main_loop_run (watchdog->main_loop);
+
+  return NULL;
+}
+
+/**
+ * @brief Create nnstreamer watchdog. Recommended using watchdog handle with proper lock (e.g., GST_OBJECT_LOCK())
+ */
+void
+nnstreamer_watchdog_create (nns_watchdog_h * watchdog_h)
+{
+  NnstWatchdog *watchdog = g_new0 (NnstWatchdog, 1);
+
+  watchdog->main_context = g_main_context_new ();
+  watchdog->main_loop = g_main_loop_new (watchdog->main_context, FALSE);
+  watchdog->thread =
+      g_thread_new ("suspend_watchdog", _nnstreamer_watchdog_thread, watchdog);
+
```

Comment:
```g_thread_try_new``` returns only if the thread is successfully created. That is, it returns NULL if the thread creation fails. On the other hand, ```g_thread_new``` always returns a pointer to the thread object. However, exception handling is required even if the thread creation fails.

Reply:
Updated.